### PR TITLE
Add cross-compilation

### LINKED ISSUE #368: Gate PRs on cross compiling successfully.
See https://discourse.mozilla-community.org/t/nightly-builds-are-back/8076 for details on the docker image. According to https://docs.travis-ci.com/user/docker/ we should be able to run this image to check that we can cross-compile.

@JohanLorenzo or @npark-mozilla can you take a look at that?

### PR DESCRIPTION
**Pull request by JohanLorenzo**
Fixes #368

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,12 +20,18 @@ matrix:
   fast_finish: true
   allow_failures:
     - rust: nightly
+  include:
+    - sudo: required
+      os: linux
+      services: docker
```
Comment (fabricedesre) [inline .travis.yml]: do we still need that?

Comment (JohanLorenzo) [inline .travis.yml]: That's correct, we don't. Thanks

**Comment by JohanLorenzo** [comment]
Green builds. The new job only compiles (and doesn't package or run any test). r? @fabricedesre

File: .cargo/config [hunk 2]
```diff
@@ -1 +1,4 @@
 paths = ["./iron-fork"]
+
+[target.armv7-unknown-linux-gnueabihf]
+linker = "arm-linux-gnueabihf-gcc"
```
Comment (fabricedesre) [inline .cargo/config]: let's not commit that, but create a $HOME/.cargo/config in travis instead.

File: tools/travis-linux-cross_compile.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+install_dependencies() {
+    _install_arm_packages
+    _set_up_environment
+    _install_rust_for_arm
+}
+
+build() {
+    cargo build --target=armv7-unknown-linux-gnueabihf
```
Comment (fabricedesre) [inline tools/travis-linux-cross_compile.sh]: nit: nightlies are built with --release. Ideally we should build both debug (the default one) and release.

**Comment by JohanLorenzo** [comment]
Changes addressed. PR rebased r? @fabricedesre

File: tools/travis-linux-arm_cross_compile.sh [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    sudo tee -a /etc/apt/sources.list <<EOF
+deb [arch=armhf,arm64] http://ports.ubuntu.com/ trusty main
+deb [arch=armhf,arm64] http://ports.ubuntu.com/ trusty-updates main
+deb [arch=armhf,arm64] http://ports.ubuntu.com/ trusty universe
+deb [arch=armhf,arm64] http://ports.ubuntu.com/ trusty-updates universe
+EOF
+
+    sudo dpkg --add-architecture armhf
+    # XXX Network issues might make the package discovery fail. If it impacts
+    # one of our packages, we will notice it at installion time
```
Comment (fabricedesre) [inline tools/travis-linux-arm_cross_compile.sh]: nit: s/installion/installation

File: tools/travis-linux-arm_cross_compile.sh [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+    tee -a $HOME/.cargo/config << EOF
+[target.armv7-unknown-linux-gnueabihf]
+linker = "arm-linux-gnueabihf-gcc"
+EOF
+}
+
+_install_rust_for_arm() {
+    # Remove preinstalled rust
+    sudo rm -rf ~/rust
```
Comment (fabricedesre) [inline tools/travis-linux-arm_cross_compile.sh]: Why do you need `sudo` here? we run `rustup.sh` with `--disable-sudo` so that should work without.

**Comment by fabricedesre** [comment]
Thanks! r=me with the last nits addressed.

File: tools/travis-linux-arm_cross_compile.sh (hunk 5 above)
Comment (JohanLorenzo) [inline tools/travis-linux-arm_cross_compile.sh]: You're right, we don't need it. Thanks for pointing it out

**Pull request merged**